ocsp_status.go

```diff
@@ -2,33 +2,42 @@
 package main
 
 import (
-	"bytes"
 	"crypto"
+	"crypto/x509"
 	"flag"
 	"fmt"
 	"github.com/googleinterns/ocsp-response-linter/linter"
 	"github.com/googleinterns/ocsp-response-linter/ocsptools"
 	"github.com/googleinterns/ocsp-response-linter/ocsptools/helpers"
 	"golang.org/x/crypto/ocsp"
-	"log"
 	"net/http"
+	"strings"
 )
 
 // checkFromFile takes a path to an OCSP Response file and then reads, parses, and lints it
-func checkFromFile(tools ocsptools.ToolsInterface, linter linter.LinterInterface, respFile string) error {
+func checkFromFile(tools ocsptools.ToolsInterface, linter linter.LinterInterface, respFile string, issuerFile string, verbose bool) error {
 	ocspResp, err := tools.ReadOCSPResp(respFile)
 	if err != nil {
 		return err
 	}
 
-	linter.LintOCSPResp(ocspResp)
+	var issuerCert *x509.Certificate // pointers initialize to nil
+
+	if issuerFile != "" {
+		issuerCert, err = tools.ParseCertificateFile(issuerFile)
+		if err != nil {
+			return fmt.Errorf("Error parsing certificate from certificate file: %w", err)
+		}
+	}
+
+	linter.LintOCSPResp(ocspResp, issuerCert, verbose)
 
 	return nil
 }
 
 // checkFromCert takes a path to an ASN.1 DER encoded certificate file and
 // constructs and sends an OCSP request then parses and lints the OCSP response
-func checkFromCert(tools ocsptools.ToolsInterface, linter linter.LinterInterface, certFile string, isPost bool, ocspURL string, dir string, hash crypto.Hash) error {
+func checkFromCert(tools ocsptools.ToolsInterface, linter linter.LinterInterface, certFile string, issuerFile string, isPost bool, ocspURL string, dir string, hash crypto.Hash, verbose bool) error {
```

Comment:
I would think checkFromFile would open the cert in the file and just call checkFromCert. In which case you wouldn't need the issuerFile here, right? Or maybe I'm misunderstanding the variables.

Reply:
So the issuerFile is a new flag that the user can use if they want to provide their own issuer certificate file. `checkFromFile` takes in an OCSP response file, so it wouldn't need the functionality within `checkFromCert` that actually fetches the OCSP response file.

Reply:
I see, yeah I think it would be good to move that out into a helper function, that shouldn't break things hopefully!